### PR TITLE
Shipping labels: Address suggestions

### PR DESCRIPTION
**Pull request by 0nko**
Implements #3019. 

Unit tests to be added in a subsequent PR.

image

**To test:**
1. Prepare an order on the web that satisfies SL creation preconditions and make sure the origin and shipping addresses are valid (but not yet validated)
2. Go to Orders -> Open the order
2. Tap on Create shipping label button
3. Tap on the Continue button
4. Notice that the address suggestion screen is displayed, showing the original and a suggested address
5. Select the entered address and tap on Edit selected address button
6. Notice the address edit screen is displayed with the address
7. Tap on Done toolbar button
8. Notice the suggestion screen is shown again
9. Select the suggested address
10. Tap on Use selected address button
11. Notice the address is accepted and displayed in the creation flow

### CODE REVIEW

**Review by hichamboushaba — COMMENTED** [review]
Great work @0nko, the code is really nice.

But I found one small issue, when I try to edit an already validated address (same as the one suggested), I still get the suggestion screen, but I don't get it in wp-admin.

I tried to track the issue, and I think the cause is in this line: https://github.com/woocommerce/woocommerce-android/blob/1ddb014a48bbe297fb76d2d4321a306df60434e2/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressFragment.kt#L361

As the backend expects the short abbreviation of the state (at least for a US address), but we send the full name, so it suggests another change to use the short one, but for the user, he didn't make any change, so the suggestion is not really clear, below a GIF showing the flow:
suggest_flow

And below the body of the different request done on the gif above: 
1. `"{"address":{"address":"60 29TH ST # 343","address_2":"","city":"SAN FRANCISCO","company":"","country":"US","name":"Hicham Boushaba","phone":"","postcode":"94110-4929","state":"CA"},"type":"destination"}"`
2. `"{"address":{"address":"60 29TH ST # 343","address_2":"","city":"SAN FRANCISCO","company":"","country":"US","name":"Hicham Boushaba ","phone":"","postcode":"94110-4929","state":"California"},"type":"destination"}"`

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModel.kt [hunk 1]
```diff
@@ -219,7 +235,8 @@ class EditShippingLabelAddressViewModel @AssistedInject constructor(
     data class ViewState(
         val address: Address? = null,
         val bannerMessage: String? = null,
-        val isProgressDialogVisible: Boolean? = null,
+        val isValidationProgressDialogVisible: Boolean? = null,
+        val isLoadingProgressDialogVisible: Boolean? = null,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModel.kt:239]: Regarding this change, as I needed a progress dialog for packages too, I added the following class in my PR: https://github.com/woocommerce/woocommerce-android/blob/f9808f6810be09f79d5fcc427dc3e8f93e5ede6a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt#L296-L300

WDYT about this approach? You don't have to change anything in your PR, if you like this approach, I can update it when I fix the conflicts.

Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/EditShippingLabelAddressViewModel.kt:239]: Oh yeah, that's perfect 👍. Thanks!

**Comment by 0nko** [comment]
Thanks for the review @hichamboushaba! I've fixed the problem and the state code is now used, as expected, which fixes the repeated suggestion issue.

**Review by hichamboushaba — APPROVED** [review]
Nice :shipit:

**Pull request merged**